Yosemite/Yosemite/Stores/SiteStore.swift

```diff
@@ -112,3 +125,27 @@ public enum SiteCreationError: Error, Equatable {
         }
     }
 }
+
+public enum SiteLaunchError: Error, Equatable {
+    case alreadyLaunched
+    case unexpected(description: String)
+
+    init(remoteError: Error) {
+        switch remoteError {
+        case let remoteError as WordPressApiError:
+            switch remoteError {
+            case let .unknown(code, _):
+                switch code {
+                case "already-launched":
+                    self = .alreadyLaunched
+                default:
+                    self = .unexpected(description: remoteError.localizedDescription)
+                }
+            default:
+                self = .unexpected(description: remoteError.localizedDescription)
+            }
+        default:
+            self = .unexpected(description: remoteError.localizedDescription)
+        }
+    }
```

Comment:
I am wondering whether we can simplify this like the following. 
```suggestion
        init(remoteError: Error) {
        guard let error = remoteError as? WordPressApiError,
              case let .unknown(code, _) = error,
              code == "already-launched" else {
            self = .unexpected(description: remoteError.localizedDescription)
            return
        }
        self = .alreadyLaunched
    }
```